parsimonious/expressions.py

```diff
@@ -272,7 +273,7 @@ def _uncached_match(self, text, pos, cache, error):
     def _regex_flags_from_bits(self, bits):
         """Return the textual equivalent of numerically encoded regex flags."""
         flags = 'tilmsux'
-        return ''.join(flags[i] if (1 << i) & bits else '' for i in xrange(6))
+        return ''.join(flags[i] if (1 << i) & bits else '' for i in range(6))
```

Comment:
> … also, looking at that code, this should be len(flags), not 6 …

I think it should be `range(1, len(flags))`. `t` isn't a valid regex flag (and neither is `0b1`, so it never gets included). I think it's impossible to get `x` included:
```python
>>> Regex('foo', verbose=True)._as_rhs()
'~"foo"'
>>> Regex('foo', ignore_case=True)._as_rhs()
'~"foo"i'
```
Fixed in 115539cbb590b47c2d64aae8d66209150e723709, so now the options display correctly:
```
>>> Regex('foo', ignore_case=True, locale=True,
...                  multiline=True, dot_all=True, unicode=True, verbose=True)._as_rhs()
'~"foo"ilmsux'
```